pkg/controller/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -0,0 +1,251 @@
+					}
+				}
+			}
+		}
+	}
+	logger.Info("done with creating TemplateUpdateRequest resources after update of NSTemplateTier", "tier", instance.Name)
+	return reconcile.Result{}, nil
+}
+
+// newLabelSelector creates a label selector to find MasterUserRecords which are not up-to-date with
+// the templateRefs of the given NSTemplateTier.
+//
+// (longer explanation)
+// newLabelSelector creates a selector to find MasterUserRecords with the following labels:
+// - toolchain.dev.openshift.com/<member>-templates-tier: <tier>
+// - toolchain.dev.openshift.com/<member>-templates-tier-hash: !<hash>
+//
+// In other words, this label selector will be used to list MasterUserRecords which have a user account set to the given `<tier>`
+// but with a template version (defined by `<hash>`) which is NOT to the expected value (the one provided by `instance`).
+//
+// Note: The `hash` value is computed from the TemplateRefs. See `computeTemplateRefsHash()`
+func newLabelSelector(instance toolchainv1alpha1.NSTemplateTier, member cluster.FedCluster) (client.MatchingLabelsSelector, error) {
+	// compute the hash of the `.spec.namespaces[].templateRef` + `.spec.clusteResource.TemplateRef`
+	hash, err := computeTemplateRefsHash(instance)
+	if err != nil {
+		return client.MatchingLabelsSelector{}, err
+	}
+	selector := labels.NewSelector()
+	tierLabel, err := labels.NewRequirement(toolchainv1alpha1.LabelKeyPrefix+member.Name+"-templates-tier", selection.Equals, []string{instance.Name})
+	selector = selector.Add(*tierLabel)
+	templateHashLabel, err := labels.NewRequirement(toolchainv1alpha1.LabelKeyPrefix+member.Name+"-templates-tier-hash", selection.NotEquals, []string{hash})
+	selector = selector.Add(*templateHashLabel)
+	return client.MatchingLabelsSelector{
+		Selector: selector,
+	}, nil
+}
+
+func (r *ReconcileNSTemplateTier) activeTemplateUpdateRequests(instance toolchainv1alpha1.NSTemplateTier) (int, error) {
+	// fetch the list of TemplateUpdateRequest owned by the NSTemplateTier instance
+	templateUpdateRequests := toolchainv1alpha1.TemplateUpdateRequestList{}
+	if err := r.client.List(context.Background(), &templateUpdateRequests, client.MatchingLabels{
+		toolchainv1alpha1.NSTemplateTierNameLabelKey: instance.Name,
+	}); err != nil {
+		return -1, err
+	}
+
+	// count non-deleted templateUpdateRequest items
+	activeTemplateUpdateRequests := 0
+	for _, r := range templateUpdateRequests.Items {
+		if r.DeletionTimestamp == nil {
+			activeTemplateUpdateRequests++
+		}
+	}
+	return activeTemplateUpdateRequests, nil
+}
+
+// compute the hash of the `.spec.namespaces[].templateRef` + `.spec.clusteResource.TemplateRef`
+func computeTemplateRefsHash(instance toolchainv1alpha1.NSTemplateTier) (string, error) {
+	spec, err := json.Marshal(instance.Spec)
+	if err != nil {
+		return "", err
+	}
+	md5hash := md5.New()
+	// Ignore the error, as this implementation cannot return one
+	md5hash.Write([]byte(spec))
```

Comment:
Let's ignore the errors explicitly. I'm still hopping we will use some linter to fetch error handling errors in the future. And that would make the linter happy ;)
```suggestion
	_, _ = md5hash.Write([]byte(spec))
```

Reply:
ok, let me apply this suggestion.

Reply:
done in 31ef427